docs/user-guide/nginx-configuration/annotations.md

```diff
@@ -43,6 +43,7 @@ You can add these Kubernetes annotations to specific Ingress objects to customiz
 |[nginx.ingress.kubernetes.io/limit-connections](#rate-limiting)|number|
 |[nginx.ingress.kubernetes.io/limit-rps](#rate-limiting)|number|
 |[nginx.ingress.kubernetes.io/permanent-redirect](#permanent-redirect)|string|
+|[nginx.ingress.kubernetes.io/permanent-redirect-code](#permanent-redirect-code)|number|
```

Comment:
The type is not aligned with `GetStringAnnotation()` which you use later. But I think `number` is actually more suitable anyway (see comments below).